x-pack/plugins/features/common/feature.ts

```diff
@@ -98,7 +98,17 @@ export interface Feature<
    * ```
    * @see FeatureKibanaPrivileges
    */
-  privileges: TPrivileges;
+  privileges:
```

Comment:
This felt more explicit to me, and also gave a couple benefits:
- Removed the generic type requirement, as it wasn't all that useful
- Simplified the `feature_schema` validation for determining if sub-features are allowed to be registered.